files/ansible/deploy_benchmarks.yml

```diff
@@ -0,0 +1,85 @@
+---
+- hosts: target
+  tasks:
+        - include_tasks: ./tasks/install_packages.yml
+          vars:
+            packages_to_install:
+              - python3
+              - python3-pip
+              - git
+              - build-essential
+              - linux-perf
+              - unzip
+              - lhasa
+
+        - name: Wipe harness directory
+          file:
+              path: "benchmark_harness"
+              state: "{{ item }}"
+          with_items:
+              - absent
+              - directory
+
+        - name: Get the benchmark_harness
+          git:
+              repo: https://github.com/Linaro/benchmark_harness.git
+              dest: benchmark_harness
+              version: "{{ branch | default('master') }}"
+
+        - name: Install pip requirements
+          shell: python3 -m pip install -r ./benchmark_harness/requirements.txt
+
+        - name: Wipe results directory
+          file:
+              path: "results"
+              state: "{{ item }}"
+          with_items:
+              - absent
+              - directory
+
+        - include_tasks: ./tasks/call_harness.yml
+
+        - name: Find where the harness put the results
+          find:
+            paths: "results"
+            patterns: '{{ benchmark }}*'
+            file_type: directory
+          register: benchresults_path
+        - debug: var=benchresults_path.files
+
+        - name: Find results
+          find:
+            paths: "{{ item.path }}/results/"
+            patterns: "^.*?\\.(?:err|out)$"
+            use_regex: yes
+          with_items: "{{ benchresults_path.files }}"
+          register: reports_path
+          failed_when: reports_path.matched == 0
+        - debug: var=reports_path
+
+        - name: Make a directory with a nice name for sftp
```

Comment:
I don't understand why you copy them to a local directory before SFTP them to the server.
Should be simpler to just find the two names above and call the SFTP role with a list.